packages/database/src/migrations/20210618033026-PGStriveAverageMortalityMatrixTable-modifies-data.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+import { generateId, insertObject, arrayToDbString } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const permissionGroupNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM permission_group WHERE name = '${name}'`);
+  return record.rows[0] && record.rows[0].id;
+};
+
+const REPORT = {
+  id: generateId(),
+  code: 'PG_Strive_Mosquito_Mortality',
+  config: {
+    fetch: {
+      dataGroups: ['STRVEC_AE-IR', 'STRVEC_AN-IR'],
+      aggregations: [
+        {
+          type: 'RAW',
+          config: {
+            dataSourceEntityType: 'facility',
+          },
+        },
+      ],
+      dataElements: ['STRVEC_AE-IR09', 'STRVEC_AE-IR02', 'STRVEC_AN-IR09', 'STRVEC_AN-IR02'],
+    },
+    output: {
+      type: 'matrix',
+      rowField: 'insecticide',
+      columns: '*',
+      categoryField: 'mosquitoType',
+    },
+    transform: [
+      {
+        '...': ['STRVEC_AE-IR09', 'STRVEC_AN-IR09', 'orgUnitName'],
+        transform: 'select',
+        "'insecticide'":
+          "exists($row['STRVEC_AE-IR02']) ? $row['STRVEC_AE-IR02'] : $row['STRVEC_AN-IR02']",
+        "'mosquitoType'": "exists($row['STRVEC_AE-IR09']) ? 'Aedes IR' : 'Anopheles IR'",
+      },
+      {
+        transform: 'aggregate',
+        insecticide: 'group',
+        orgUnitName: 'group',
+        mosquitoType: 'group',
+        'STRVEC_AE-IR09': 'avg',
+        'STRVEC_AN-IR09': 'avg',
+      },
+      {
+        transform: 'select',
+        '$row.orgUnitName':
+          "exists($row['STRVEC_AE-IR09']) ? $row['STRVEC_AE-IR09'] : $row['STRVEC_AN-IR09']",
+        '...': ['insecticide', 'mosquitoType'],
+      },
+      {
+        transform: 'aggregate',
+        insecticide: 'group',
+        mosquitoType: 'group',
+        '...': 'last',
+      },
+    ],
+  },
+};
+
+const DASHBOARD_GROUP_CODES = ['PG_Strive_PNG_Country'];
+
+const BASE_DASHBOARD_REPORT = {
+  id: 'PG_Strive_Mosquito_Mortality',
+  dataBuilder: 'reportServer',
+  dataServices: [
+    {
+      isDataRegional: true,
+    },
+  ],
+  dataBuilderConfig: { reportCode: REPORT.code },
+};
+
+const VIEW_JSON = {
+  name: 'Average Mosquito Mortality',
+  type: 'matrix',
+  periodGranularity: 'one_month_at_a_time',
```

Comment:
Might be worth getting a confirmation that we want data 1 month at a time? I know it's there in the ticket screenshot but just not sure if its consistent with the other dashboards we've been building which I think have just been for all time?

Reply:
Sure, I will raise a comment.

Reply:
Thanks Rohan, we did need another date picker.